src/java/org/apache/cassandra/config/Config.java

```diff
@@ -1283,4 +1283,21 @@ public static void log(Config config)
     public double severity_during_decommission = 0;
 
     public StorageCompatibilityMode storage_compatibility_mode;
+
+    public enum CQLStartTime
+    {
+        REQUEST, // uses a timestamp that represent the start of processing of the request
+        QUEUE    // uses a timestamp that represents when the request was enqueued
+    }
+    public volatile CQLStartTime cql_start_time = CQLStartTime.REQUEST;
+
+    public boolean native_transport_throw_on_overload = false;
+    public double native_transport_queue_max_item_age_threshold = Double.MAX_VALUE;
+    public DurationSpec.LongMillisecondsBound native_transport_min_backoff_on_queue_overload = new DurationSpec.LongMillisecondsBound("10ms");
+    public DurationSpec.LongMillisecondsBound native_transport_max_backoff_on_queue_overload = new DurationSpec.LongMillisecondsBound("200ms");
+
+    // 3.0 Cassandra Driver has its "read" timeout set to 12 seconds. Our recommendation is match this. Default is 100 seconds, which is extremely high
+    // but still not completely unbounded.
+    public DurationSpec.LongMillisecondsBound native_transport_timeout = new DurationSpec.LongMillisecondsBound("100000ms");
```

Comment:
This default is likely moving to something lower, at least from what I'm reading in the Jira?  Is there any reference value in anything else on the classpath, or do we kind of just have to hard-code something w/ out-of-band knowledge. Do we want to use something like 20, 30...or go all the way to 12.

Also, which of these, if any, do we want to document in `cassandra.yaml`? I'm okay w/ leaving things out that we really don't operators should change...

Reply:
Can we just do this?

```
public DurationSpec.LongMillisecondsBound native_transport_timeout = new DurationSpec.LongMillisecondsBound(SocketOptions.DEFAULT_READ_TIMEOUT_MILLIS, TimeUnit.MILLISECONDS);
```

Even if we can't, could we reference `{@link SocketOptions.DEFAULT_READ_TIMEOUT_MILLIS}` in the JavaDoc around the patch for posterity?

Reply:
The problem is that some people expect that if their read timeout is 5 seconds, and request has spent 5 seconds in the queue, it will still be executed. Native request timeout is based off when we have admitted the request to the queue, which will make it time out after the 5 seconds in the queue. 

 I have changed the default to 12 seconds for now, which based on discussion with @driftx on Jira seems to be a consensus.